.travis.yml

```diff
@@ -1,15 +1,86 @@
+#------------------------------------------------------------------------------
+# TravisCI configuration file for solidity.
+#
+# The documentation for solidity is hosted at:
+#
+# http://solidity.readthedocs.org
+#
+# TODO - Merge with .travis.yml file which currently lives in the root of
+# webthree-umbrella, but actually contains the automation for the Solidity
+# Emscripten build, which will also need consolidating into here somehow.
+#
+# See https://github.com/ethereum/webthree-umbrella/blob/develop/.travis.yml
+#
+# (c) 2016 cpp-ethereum contributors.
+#------------------------------------------------------------------------------
+
 language: cpp
-cache: ccache
+branches:
+    only:
+        - develop
+        - release
+        - standalone
+    except:
+        - /develop-v[0-9]/
 matrix:
     include:
         - os: linux
           dist: trusty
           sudo: required
-          env: TRAVIS_BUILD_TYPE=Debug
+          env:
+              - TRAVIS_BUILD_TYPE=RelWithDebInfo
+              - ZIP_SUFFIX=ubuntu-trusty
+        - os: osx
+          osx_image: beta-xcode6.2
+          env:
+              - TRAVIS_BUILD_TYPE=RelWithDebInfo
+              - ZIP_SUFFIX=osx-mavericks
+        - os: osx
+          osx_image: xcode7.1
+          env:
+              - TRAVIS_BUILD_TYPE=RelWithDebInfo
+              - ZIP_SUFFIX=osx-yosemite
         - os: osx
           osx_image: xcode7.3
-          env: TRAVIS_BUILD_TYPE=Debug
+          env:
+              # The use of Debug config here ONLY for El Capitan is a workaround for "The Heisenbug"
+              # See https://github.com/ethereum/webthree-umbrella/issues/565
+              - TRAVIS_BUILD_TYPE=Debug
+              - ZIP_SUFFIX=osx-elcapitan
 git:
     depth: 2
-before_install: ./install_dependencies.sh
-script: ./fetch_umbrella_build_and_test.sh
\ No newline at end of file
+cache:
+    - ccache
+    - directories: build
+install:
+    - ./scripts/install_deps.sh
+before_script:
+    - mkdir -p build
+    - cd build
+    - cmake .. -DCMAKE_BUILD_TYPE=$TRAVIS_BUILD_TYPE
+    - make -j2
+    - ../scripts/release.sh $ZIP_SUFFIX
+script:
+    - ETH_PATH_FOR_linux="eth"
+    - ETH_PATH_FOR_osx="../eth"
+    - ETH_PATH_EXPRESSION_NAME=ETH_PATH_FOR_$TRAVIS_OS_NAME
+    - ETH_PATH=$ETH_PATH_EXPRESSION_NAME
+    - eval \$$ETH_PATH --test -d /tmp/test &
+    - while [ ! -S /tmp/test/geth.ipc ]; do sleep 2; done
```

Comment:
Please move this into a script. The travis file should be used for configuration, but the actual scripts should be somewhere else. The other benefit of moving this to a script would be that this can be run outside of travis.